x-pack/plugin/core/src/main/java/org/elasticsearch/snapshots/SourceOnlySnapshotRepository.java

```diff
@@ -0,0 +1,151 @@
+ * match_all scroll searches in order to reindex the data.
+ */
+public final class SourceOnlySnapshotRepository extends FilterRepository {
+    private static final Setting<String> DELEGATE_TYPE = new Setting<>("delegate_type", "", Function.identity(), Setting.Property
+        .NodeScope);
+    public static final Setting<Boolean> SOURCE_ONLY = Setting.boolSetting("index.source_only", false, Setting
+        .Property.IndexScope, Setting.Property.Final, Setting.Property.PrivateIndex);
+
+    private static final String SNAPSHOT_DIR_NAME = "_snapshot";
+
+    public static Repository.Factory newFactory() {
+        return new Repository.Factory() {
+
+            @Override
+            public Repository create(RepositoryMetaData metadata) {
+                throw new UnsupportedOperationException();
+            }
+
+            @Override
+            public Repository create(RepositoryMetaData metaData, Function<String, Repository.Factory> typeLookup) throws Exception {
+                String delegateType = DELEGATE_TYPE.get(metaData.settings());
+                if (Strings.hasLength(delegateType) == false) {
+                    throw new IllegalArgumentException(DELEGATE_TYPE.getKey() + " must be set");
+                }
+                Repository.Factory factory = typeLookup.apply(delegateType);
+                return new SourceOnlySnapshotRepository(factory.create(new RepositoryMetaData(metaData.name(),
+                    delegateType, metaData.settings()), typeLookup));
+            }
+        };
+    }
+
+    SourceOnlySnapshotRepository(Repository in) {
+        super(in);
+    }
+
+    @Override
+    public IndexMetaData getSnapshotIndexMetaData(SnapshotId snapshotId, IndexId index) throws IOException {
+        IndexMetaData snapshotIndexMetaData = super.getSnapshotIndexMetaData(snapshotId, index);
+        // TODO: can we lie about the index.version.created here and produce an index with a new version since we reindex anyway?
+
+        // for a minimal restore we basically disable indexing on all fields and only create an index
+        // that is fully functional from an operational perspective. ie. it will have all metadata fields like version/
+        // seqID etc. and an indexed ID field such that we can potentially perform updates on them or delete documents.
+        ImmutableOpenMap<String, MappingMetaData> mappings = snapshotIndexMetaData.getMappings();
+        Iterator<ObjectObjectCursor<String, MappingMetaData>> iterator = mappings.iterator();
+        IndexMetaData.Builder builder = IndexMetaData.builder(snapshotIndexMetaData);
+        while (iterator.hasNext()) {
+            ObjectObjectCursor<String, MappingMetaData> next = iterator.next();
+            MappingMetaData.Routing routing = next.value.routing();
+            final String mapping;
+            if (routing.required()) { // we have to respect the routing to be on the safe side so we pass this one on.
```

Comment:
can you clarify what you mean with respect routing? we don't store the value and get isn't supported.

Reply:
this is correct we can remove this. I will